src/linear-algebra.lisp

```diff
@@ -236,6 +236,14 @@ The function will just return NIL, and modify the contents of RESULT."
                                    :name (string (gensym "KRAUS-TEMP"))
                                    :matrix mat)))))
 
+(defun superoperator-to-matrix (sop)
+  "Convert SOP into a MAGICL:MATRIX or a LIST of them."
+  (adt:match superoperator sop
+    ((single-kraus U)
+     (quil::gate-matrix U))
+    ((kraus-list list)
```

Comment:
I don't think this is right. `kraus-list` still represents a single superoperator, so I wouldn't expect a list in return.

Reply:
(it could be that this function is simply misnamed if it's producing the preferred output for its callers)

Reply:
Probably badly named. I did consider slapping an `-or-list` on the end or some such, but didn't seem like a huge improvement. I'm open to suggestions.

This function is meant to be the inverse of [`CONVERT-TO-KRAUS-LIST` in `apply-gate.lisp`](https://github.com/rigetti/qvm/blob/ac22e422d9b172dbbd9bbec2db88acf366eddf9f/src/apply-gate.lisp#L101).

It's called in order to convert a superoperator into something that can be passed to `check-kraus` in `check-superoperators` to ensure the superoperators handed to `density-qvm` and `base-qvm` satisfy the `check-kraus` constraints.

Reply:
`CONVERT-FROM-KRAUS-LIST` (j/k)

Reply:
`I-CAN-HAZ-KRAUS-OPS?` 🐱🍔

Reply:
Or maybe `SUPEROPERATOR-TO-KRAUS-OPS`, since "kraus ops" seems to be semi-standard naming convention elsewhere in the code?  E.g. `CHECK-KRAUS-OPS`, `GENERATE-KRAUS-OPS`, etc.  Idk, I'm grabbing at straws here.

Reply:
`superoperator-to-component-matrices`? At least that gives a hint that it might not return a single matrix.

Reply:
I like it.

Reply:
tis done